### PR TITLE
ploticus 2.42 (new formula)

### PR DESCRIPTION
**Pull request by sjalander**
Ploticus is a simple tool for generating graphs

### CODE REVIEW

File: Library/Formula/ploticus242.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "http://ploticus.sourceforge.net/"
+  url "https://downloads.sourceforge.net/project/ploticus/ploticus/2.42/ploticus242_src.tar.gz?r=http%3A%2F%2Fsourceforge.net%2Fprojects%2Fploticus%2F%3Fsource%3Dtyp_redirect&ts=1449655936"
+  version "2.42"
+  sha256 "3f29e4b9f405203a93efec900e5816d9e1b4381821881e241c08cab7dd66e0b0"
+
+  def install
+    cd "src" do
+      system "make"
+      bin.install "pl"
+    end
```
Comment (bfontaine) [inline Library/Formula/ploticus242.rb]: You can probably shorten this into:
```rb
system "make", "-C", "src"
bin.install "src/pl"
```

File: Library/Formula/ploticus242.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    system "pl 2>&1 >/dev/null | grep 'usage: pl scriptfile'"
```
Comment (bfontaine) [inline Library/Formula/ploticus242.rb]: Please use `assert_match "usage: pl scriptfile", shell_output("#{bin}/pl 1>&1")`.

Comment (sjalander) [inline Library/Formula/ploticus242.rb]: The issue is that pl returns 1 so this way the test is not passed.

/Magnus

On Wed, Dec 9, 2015 at 11:43 AM, Baptiste Fontaine <notifications@github.com

Comment (bfontaine) [inline Library/Formula/ploticus242.rb]: Use `shell_output("#{bin}/pl 1>&1", 1)` then.

Comment (sjalander) [inline Library/Formula/ploticus242.rb]: Ok, fixat.

/Magnus

On Wed, Dec 9, 2015 at 12:25 PM, Baptiste Fontaine <notifications@github.com

File: Library/Formula/ploticus242.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "libpng"
+
+  def install
+    system "make", "-C", "src"
+    bin.install "src/pl"
+  end
+
+  test do
+    assert_match "usage: pl scriptfile", shell_output("#{bin}/pl 2>&1", 1)
```
Comment (dunn) [inline Library/Formula/ploticus242.rb]: It'd be great to have a test that checks the functionality more; maybe create a small graph?

Comment (sjalander) [inline Library/Formula/ploticus242.rb]: And how would you verify that the graph is correct.
Have a binary blob of an eps that you compare against?

/Magnus

Comment (mikemcquaid) [inline Library/Formula/ploticus242.rb]: That it doesn't segfault and returns a successful exit code would be enough

**Comment by bfontaine** [comment]
Ping on the test?

**Comment by sjalander** [comment]
I don't have the time to add such a test at the moment so if that is a
requirement then I guess it will not go into homebrew at this time.

/Magnus

File: Library/Formula/ploticus242.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class Ploticus242 < Formula
```
Comment (apjanke) [inline Library/Formula/ploticus242.rb]: This should probably be plain `Ploticus` and live in file `ploticus.rb`. Homebrew doesn't bake version numbers in to formula names.

Comment (sjalander) [inline Library/Formula/ploticus242.rb]: Class and filename has been updated.

/Magnus

**Comment by mikemcquaid** [comment]
Closing in favour of #47127.

**Comment by apjanke** [comment]
I've merged the revised ploticus formula.

**Pull request closed without merging**